src/CoreBundle/Resources/public/javascript/tableEditor.js

```diff
@@ -539,11 +539,13 @@ function SaveViaAjax() {
 function SaveViaAjaxCallback(data, statusText) {
     CloseModalIFrameDialog();
 
-    // remove all message-classes from the fields
-    $('*[id^="fieldname_"]').removeClass();
+    // remove all message background classes from field containers.
+    $('*[id^="fieldname_"]').removeClass(function (index, className) {
+        return (className.match (/(^|\s)bg-\S+/g) || []).join(' ');
+    });
 
     var returnVal = false;
-    if (data != false && data != null) {
+    if (data !== false && data != null) {
```

Comment:
Are you confident enough that this change won't cause problems? The true/false rules in JS are at least as wonderful and straightforward as in PHP.

Reply:
tested this. works. So, yes.